Altis_Life.Altis/dialog/function/fn_unimpound.sqf

```diff
@@ -31,6 +33,20 @@ switch (playerSide) do {
     case independent: {_purchasePrice = _price * LIFE_SETTINGS(getNumber,"vehicle_purchase_multiplier_MEDIC");};
     case east: {_purchasePrice = _price * LIFE_SETTINGS(getNumber,"vehicle_purchase_multiplier_OPFOR");};
 };
+
+if (!(_vehicleShop isEqualTo "")) then {
+    if (isClass (missionConfigFile >> "CarShops" >> _vehicleShop)) then {
+        _vehicleList = M_CONFIG(getArray,"CarShops",_vehicleShop,"vehicles");
+        _vehicleIndex = [_vehicle,_vehicleList] call TON_fnc_index;
+
+        if (!(_vehicleIndex isEqualTo -1)) then {
+            _vehiclePrice = ((_vehicleList select _vehicleIndex) select 1) select 0;
+            if (!(_vehiclePrice isEqualTo -1)) then {
+                _purchasePrice = _vehiclePrice;
+            };
+        };
+    };
+};
```

Comment:
When I said about a function, it was for this. You do this exact thing 4 times ;)